src/Controller/GmoLinkTypePlusController.php

```diff
@@ -244,16 +216,31 @@ public function updateLinkTypePaymentStatus($order_id, $payment_method, $linkTyp
         ]);
         $payment->save();
       }
-      if ($success_page) {
+      if ($this->paymentSuccess && $this->isOrderCreationNeeded) {
         $order->unlock();
         $order->setData($paymentGateway, $data);
         if ($order->getState()->getId() != 'completed') {
-          $order->getState()->applyTransitionById('place');
+          $order->getState()->applyTransitionById('fulfill');
         }
+        $order->setPlacedTime(\Drupal::time()->getCurrentTime());
+        $order->setOrderNumber($order_id);
+        $order->set('cart', 0);
         $order->save();
```

Comment:
we have to check why this is not triggering automatically. Also it may stop triggering the events and action we already have